src/xbot/common/math/PIDManager.java

```diff
@@ -13,26 +13,37 @@
     private DoubleProperty propP;
     private DoubleProperty propI;
     private DoubleProperty propD;
+    private DoubleProperty maxOutput;
+    private DoubleProperty minOutput;
 
+    public PIDManager(String functionName, XPropertyManager propMan, double defaultP, double defaultI, double defaultD,
+            double defaultMaxOutput, double defaultMinOutput) {
+        setupPIDManager(functionName, propMan, defaultP, defaultI, defaultD, defaultMaxOutput, defaultMinOutput);
+    }
+    
     public PIDManager(String functionName, XPropertyManager propMan, double defaultP, double defaultI, double defaultD) {
-        setupPIDManager(functionName, propMan, defaultP, defaultI, defaultD);
+        setupPIDManager(functionName, propMan, defaultP, defaultI, defaultD, 1.0, -1.0);
     }
 
     public PIDManager(String functionName, XPropertyManager propMan) {
-        setupPIDManager(functionName, propMan, 0, 0, 0);
+        setupPIDManager(functionName, propMan, 0, 0, 0, 1.0, -1.0);
     }
 
     private void setupPIDManager(String functionName, XPropertyManager propMan, double defaultP, double defaultI,
-            double defaultD) {
-        propP = new DoubleProperty(functionName + " P", defaultP, propMan);
-        propI = new DoubleProperty(functionName + " I", defaultI, propMan);
-        propD = new DoubleProperty(functionName + " D", defaultD, propMan);
+            double defaultD, double defaultMaxOutput, double defaultMinOutput) {
+        propP = propMan.createPersistentProperty(functionName + " P", defaultP);
+        propI = propMan.createPersistentProperty(functionName + " I", defaultI);
+        propD = propMan.createPersistentProperty(functionName + " D", defaultD);
+        
+        maxOutput = propMan.createPersistentProperty(functionName + " MaxOutput", defaultMaxOutput);
```

Comment:
I'm not sure we ever really settled on a philosophy for what things should look like in the property names. I feel like most of the robot ones are camel cased for instance.